src/ring_theory/ideal/basic.lean

```diff
@@ -590,6 +590,18 @@ not_is_field_iff_exists_ideal_bot_lt_and_lt_top.trans
     ⟨p, bot_lt_iff_ne_bot.mp (lt_of_lt_of_le bot_lt le_p), hp.is_prime⟩,
    λ ⟨p, ne_bot, prime⟩, ⟨p, bot_lt_iff_ne_bot.mpr ne_bot, lt_top_iff_ne_top.mpr prime.1⟩⟩
 
+lemma is_field_iff_forall_ideal_eq [nontrivial R] :
+  is_field R ↔ ∀ I : ideal R, I = ⊥ ∨ I = ⊤ :=
+begin
+  rw [← not_iff_not, ring.not_is_field_iff_exists_ideal_bot_lt_and_lt_top],
+  push_neg,
+  simp_rw [lt_top_iff_ne_top, bot_lt_iff_ne_bot],
+end
+
+lemma _root_.field.ideal_eq {R : Type*} [field R] (I : ideal R) :
+  I = ⊥ ∨ I = ⊤ :=
+ring.is_field_iff_forall_ideal_eq.mp (field.to_is_field R) I
```

Comment:
I think `is_simple_order.eq_bot_or_eq_top I` is enough to prove this, suggesting the lemma isn't that useful